### DISCUSSION TITLE
JSON SDK - User not set

### DISCUSSION THREAD

**Discussion by haIIux** [TelemetryDeck/Viewer#271] [Q&A] [answered]
Forgive me as I am no web developer, and I don't deal with JS. I have a static website which is strictly HTML, no CSS or JS. I want to use TelemetryDeck simply to get some analyitics on visitors so I assume that the page load example would be sufficient. To do so, I snatched the below code and swapped in my app ID however, I'm getting no signals and the following error in my console...
> Can't find variable: USER_IDENTIFIER
```
<script src="https://unpkg.com/@telemetrydeck/sdk/dist/telemetrydeck.min.js" defer></script>
<script>
  window.td = window.td || [];
  td.push(["app", MYAPPID], ["user", USER_IDENTIFIER], ["signal"]);
</script>
```

It's my understanding, which may be wrong that if I don't specify `user` it will be anon. Which is fine, but does that mean that I eliminate the `["user", USER_IDENTIFIER]` all together? When I attempted I got the following error...

> Error: TelemetryDeck: "user" is not set

If I keep `["user"]` in, I get the same error.

**Comment by haIIux** [comment, marked as answer]
This appears to be my misunderstanding with the documentation. The following code fixed my issue, I apologize.

```js
<script src="https://unpkg.com/@telemetrydeck/sdk/dist/telemetrydeck.min.js" defer></script>
<script>
  window.td = window.td || [];
  td.push(["app", MYAPPID], ["user", "anon"], ["signal"]);
</script>
```

**Comment by winsmith** [comment]
Sorry for the confusion, your answer is the correct one. We're rethinking how the web sdk should work at the moment and I'm confident this will be easier soon.
